action.yml

```diff
@@ -36,43 +36,107 @@ outputs:
       shell: 'bash'
       id: 'oxygenctl-action'
-      if: env.IS_V2 == 'false'
+      if: always()
       env:
         OXYGEN_ASSETS_DIR: ${{ inputs.oxygen_client_dir }}
         OXYGEN_BUILD_COMMAND: ${{ inputs.build_command }}
         OXYGEN_COMMIT_MESSAGE: ${{ inputs.commit_message || 'No commit message' }}
-        OXYGEN_DEPLOYMENT_TOKEN: ${{ inputs.oxygen_deployment_token }}
         OXYGEN_WORKFLOW_ID: "${{ github.run_id }}_${{ github.run_attempt }}"
         OXYGEN_WORKER_DIR: ${{ inputs.oxygen_worker_dir }}
       run: |
-        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}";
-        [[ -z "${{ inputs.commit_timestamp }}" ]] && export OXYGEN_COMMIT_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ") || export OXYGEN_COMMIT_TIMESTAMP=${{ inputs.commit_timestamp }};
-        echo "url=$($GITHUB_ACTION_PATH/oxygenctl deploy --health-check='${{ inputs.oxygen_health_check }}')" >> $GITHUB_OUTPUT
+        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}"
+
+        if [ "${{ env.IS_V2 }}" == "true" ]; then
+          echo "Deploying..."
+          mkdir oxygen_v2
+          yarn global add @shopify/oxygen-cli@latest
+          rsync -a ./* ./oxygen_v2/ --exclude 'oxygen_v2' --exclude 'node_modules'
+
+          oxygen_v2_command="oxygen-cli oxygen:deploy \
+                --path=${{ inputs.path }} \
+                --assetsFolder=${{ inputs.oxygen_client_dir }} \
+                --workerFolder=${{ inputs.oxygen_worker_dir }} \
+                --token=${{ steps.check_version.outputs.v2_token }}"
+
+          if [[ "${{ env.PRIORITY }}" == "v2" ]]; then
+            echo "v2_url=$(cd oxygen_v2 && ln -s ../node_modules node_modules && $oxygen_v2_command --buildCommand='${{ inputs.build_command }}')" >> v2_output &
+          else
+            echo "v2_url=$(cd oxygen_v2 && ln -s ../node_modules node_modules && $oxygen_v2_command --buildCommand='${{ inputs.build_command }}' &> /dev/null)" >> v2_output &
+          fi
+        fi
+
+        if [ "${{ env.IS_V1 }}" == "true" ]; then
+          echo "Deploying..."
+          [[ -z "${{ inputs.commit_timestamp }}" ]] && export OXYGEN_COMMIT_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ") || export OXYGEN_COMMIT_TIMESTAMP=${{ inputs.commit_timestamp }}
+
+          oxygen_v1_command="$GITHUB_ACTION_PATH/oxygenctl deploy --health-check=${{ inputs.oxygen_health_check }}"
+
+          if [[ "${{ env.PRIORITY }}" == "v1" ]]; then
+            echo "v1_url=$(OXYGEN_DEPLOYMENT_TOKEN=${{ steps.check_version.outputs.v1_token }} $oxygen_v1_command)" >> v1_output &
+          else
+            echo "v1_url=$(OXYGEN_DEPLOYMENT_TOKEN=${{ steps.check_version.outputs.v1_token }} $oxygen_v1_command &> /dev/null)" >> v1_output &
+          fi
+        fi
+
+        wait
+
+        if [[ "${{ env.IS_V2 }}" == "true" && "${{ env.PRIORITY }}" == "v2" ]]; then
+          cat v2_output
+          echo "url=$(grep -oP 'v2_url=\K.*' v2_output)" >> $GITHUB_OUTPUT
+        else
+          cat v1_output
```

Comment:
I think we should these `cat`'s as well? The cli will display the preview url anyway